drivers/soundwire/bus.c

```diff
@@ -1280,6 +1280,12 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
 			"SDW_SCP_INTMASK1 write failed:%d\n", ret);
 		return ret;
 	}
+	/* Clear bus clash interrupt */
+	ret = sdw_update_no_pm(slave, SDW_SCP_INT1,
+			       SDW_SCP_INT1_BUS_CLASH, SDW_SCP_INT1_BUS_CLASH);
+	if (ret < 0)
+		dev_err(&slave->dev,
+			"Clear SDW_SCP_INT1_BUS_CLASH failed:%d\n", ret);
```

Comment:
How about doing the opposite, first clear the status and then enable the interrupt?

That would remove any memory of the initial bus clash, but allow for programming errors or persistent electrical errors to be detected.

I also think this should be an optional bus behavior, e.g. for Intel devices this behavior would be the default but we don't force all implementations to behave the same.

The commit message is also questionable, a bus clash is not a harmless error!

Reply:
> How about doing the opposite, first clear the status and then enable the interrupt?
> 
> That would remove any memory of the initial bus clash, but allow for programming errors or persistent electrical errors to be detected.

I tested it, but clearing before enabling doesn't help. I can still see bus clash on my side. On the other hand, clearing right after enabling fixes the clash issue on my side.

> 
> I also think this should be an optional bus behavior, e.g. for Intel devices this behavior would be the default but we don't force all implementations to behave the same.
> 
> The commit message is also questionable, a bus clash is not a harmless error!

Agree.
I will make it optional and revise the commit message.

Reply:
> > How about doing the opposite, first clear the status and then enable the interrupt?
> > That would remove any memory of the initial bus clash, but allow for programming errors or persistent electrical errors to be detected.
> 
> I tested it, but clearing before enabling doesn't help. I can still see bus clash on my side. On the other hand, clearing right after enabling fixes the clash issue on my side.
> 

Sorry, I tested it with a wrong way. I tested it with ``sdw_write_no_pm``. i.e. without reading operation. That's why ``bus clash`` wasn't cleared. The test result is good if I tested it with the updated version.

Reply:
clear register is not the key step, read register is. It is very trick. @bardliao do you plan to disclose this idea ?